forge-game/src/main/java/forge/game/card/CardFactoryUtil.java

```diff
@@ -1255,6 +1225,22 @@ public static void addTriggerAbility(final KeywordInterface inst, final Card car
 
             saRebel.setIntrinsic(intrinsic);
             inst.addTrigger(etbTrigger);
+        } else if (keyword.equals("Gift")) {
```

Comment:
The ETB trigger only makes sense for Permanents, so i think we could skip them for others?

Reply:
Yea i know. I wasn't done with this part yet.